src/server/session.ts

```diff
@@ -1199,6 +1201,49 @@ namespace ts.server {
             }
         }
 
+        private getCodeFixes(args: protocol.CodeFixRequestArgs, simplifiedResult: boolean): protocol.CodeAction[] | CodeAction[] {
+            const { file, project } = this.getFileAndProjectWithoutRefreshingInferredProjects(args);
+
+            const scriptInfo = project.getScriptInfoForNormalizedPath(file);
+            const startPosition = getStartPosition();
+            const endPosition = getEndPosition();
+
+            const codeActions = project.getLanguageService().getCodeFixesAtPosition(file, startPosition, endPosition, args.errorCodes);
+            if (!codeActions) {
+                return undefined;
+            }
+            if (simplifiedResult) {
+                return codeActions.map(mapCodeAction);
+            }
+            else {
+                return codeActions;
+            }
+
+            function mapCodeAction(source: CodeAction): protocol.CodeAction {
+                return {
+                    description: source.description,
+                    changes: source.changes.map(change => ({
+                        fileName: change.fileName,
+                        textChanges: change.textChanges.map(textChange => ({

```

Comment:
extract the common part between here and `getFormattingEditsForRange` in `convertTextChangeToCodeEdit`